tests/integration/aiplatform/test_dataset.py

```diff
@@ -0,0 +1,278 @@
+        yield
+
+    @pytest.fixture()
+    def create_image_dataset(self, dataset_gapic_client, shared_state):
+
+        gapic_dataset = dataset.Dataset(
+            display_name=f"temp_sdk_integration_test_create_image_dataset_{uuid.uuid4()}",
+            metadata_schema_uri=aiplatform.schema.dataset.metadata.image,
+        )
+
+        create_lro = dataset_gapic_client.create_dataset(
+            parent=_TEST_PARENT, dataset=gapic_dataset
+        )
+        new_dataset = create_lro.result()
+        shared_state["dataset_name"] = new_dataset.name
+        yield
+
+    @pytest.fixture()
+    def delete_new_dataset(self, dataset_gapic_client, shared_state):
+        yield
+        assert shared_state["dataset_name"]
+
+        deletion_lro = dataset_gapic_client.delete_dataset(
+            name=shared_state["dataset_name"]
+        )
+        deletion_lro.result()
+
+        shared_state["dataset_name"] = None
+
+    def test_get_existing_dataset(self):
+        """Retrieve a known existing dataset, ensure SDK successfully gets the
+        dataset resource."""
+
+        aiplatform.init(project=_TEST_PROJECT, location=_TEST_LOCATION)
+
+        flowers_dataset = aiplatform.Dataset(dataset_name=_TEST_IMAGE_DATASET_ID)
+        assert flowers_dataset.name == _TEST_IMAGE_DATASET_ID
+        assert flowers_dataset.display_name == _TEST_DATASET_DISPLAY_NAME
+
+    def test_get_nonexistent_dataset(self):
+        """Ensure attempting to retrieve a dataset that doesn't exist raises
+        a Google API core 404 exception."""
+
+        aiplatform.init(project=_TEST_PROJECT, location=_TEST_LOCATION)
+
+        # AI Platform service returns 404
+        with pytest.raises(exceptions.NotFound):
+            aiplatform.Dataset(dataset_name="0")
+
+    @pytest.mark.usefixtures("create_text_dataset", "delete_new_dataset")
+    def test_get_new_dataset_and_import(self, dataset_gapic_client, shared_state):
+        """Retrieve new, empty dataset and import a text dataset using import().
+        Then verify data items were successfully imported."""
+
+        assert shared_state["dataset_name"]
+        aiplatform.init(project=_TEST_PROJECT, location=_TEST_LOCATION)
+
+        my_dataset = aiplatform.Dataset(dataset_name=shared_state["dataset_name"])
+
+        data_items_pre_import = dataset_gapic_client.list_data_items(
+            parent=my_dataset.resource_name
+        )
+
+        assert len(list(data_items_pre_import)) == 0
+
+        # Blocking call to import
+        my_dataset.import_data(
+            gcs_source=_TEST_TEXT_ENTITY_EXTRACTION_GCS_SOURCE,
+            import_schema_uri=_TEST_TEXT_ENTITY_IMPORT_SCHEMA,
+        )
+
+        data_items_post_import = dataset_gapic_client.list_data_items(
+            parent=my_dataset.resource_name
+        )
+
+        assert len(list(data_items_post_import)) > 400  # Expected value 469
+
+    @pytest.mark.usefixtures("delete_new_dataset")
```

Comment:
Confirming this gets called if the test fails internally?

Reply:
It does not, I'll look into capturing the `AssertionError` and allowing the teardown to still proceed.

Reply:
Resolved this by setting `shared_state['dataset_name']` before making the assertions